MosipNexus/crawler/web_crawler.py

```
@@ -0,0 +1,210 @@
+                    headers=HTTP_HEADERS,
+                    timeout=30,
+                ).text,
+                "html.parser",
+            )
+            for a in soup.find_all("a", href=True):
+                next_url = urljoin(url, str(a["href"])).split("#")[0]
+                cand = urlparse(next_url)
+                base = urlparse(base_url)
+                if cand.scheme in {"http", "https"} and cand.netloc == base.netloc:
+                    _crawl(next_url, d - 1)
+        except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)